src/CodeGen_C.cpp

```diff
@@ -1300,7 +1311,12 @@ string CodeGen_C::print_reinterpret(Type type, const Expr &e) {
     } else {
         oss << "reinterpret<" << print_type(type) << ">";
     }
-    oss << "(" << print_expr(e) << ")";
+    if (type.is_handle() && !is_const_pointer(type) &&
+        is_const_pointer(e.type())) {
+        oss << "const_cast<std::add_pointer<std::remove_const<std::remove_pointer<decltype(" << print_expr(e) << ")>::type>::type>::type>(" << print_expr(e) << ")";
```

Comment:
..."The Aristocrats"!

Reply:
Had to come up with another answer anyway...